src/main/java/com/iota/iri/network/FIFOCache.java

```diff
@@ -0,0 +1,79 @@
+package com.iota.iri.network;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.SecureRandom;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * The {@link FIFOCache} is a simple FIFO cache which removes entries at the front of the queue when the capacity is
+ * reached. The {@link FIFOCache} might also randomly drop entries defined as per the given drop rate.
+ * 
+ * @param <K> the key type
+ * @param <V> the value type
+ */
+public class FIFOCache<K, V> {
+
+    private static final Logger log = LoggerFactory.getLogger(FIFOCache.class);
+
+    private ReadWriteLock cacheLock = new ReentrantReadWriteLock(true);
+    private final int capacity;
+    private final double dropRate;
+    private LinkedHashMap<K, V> map;
+    private final SecureRandom rnd = new SecureRandom();
+
+    /**
+     * Creates a new {@link FIFOCache}.
+     * 
+     * @param capacity the maximum capacity of the cache
+     * @param dropRate the rate at which to randomly drop entries
+     */
+    public FIFOCache(int capacity, double dropRate) {
```

Comment:
As I wrote in the issue we can move to a standard cache implementation and not have to maintain code.
No need to do it now.
Just fix the constructor please